tools/projmgr/src/ProjMgrKernel.cpp

```diff
@@ -127,6 +127,15 @@ bool ProjMgrKernel::LoadAndInsertPacks(std::list<RtePackage*>& packs, std::list<
   }
 
   globalModel->InsertPacks(newPacks);
-  packs.insert(packs.end(), newPacks.begin(), newPacks.end());
+
+  // Only append packs that were actually insterted into the model
+  for (const auto& pack : newPacks) {
+    const string& packId = pack->GetPackageID();
+    if (globalModel->GetPackage(packId) == pack) {
+      if (find(packs.cbegin(), packs.cend(), pack) == packs.cend()) {
+        packs.push_back(pack);
+      }
+    }
+  }
```

Comment:
To ensure the proper model inserted packs are tracked as loaded packs, I would suggest to use the `GetPackages()` function:
```suggestion
  // Track only packs that were actually inserted into the model
  packs.clear();
  for (const auto& [_, pack] : globalModel->GetPackages()) {
    packs.push_back(pack);
  }
```